Lib/fontTools/ttLib/tables/_k_e_r_n.py

```diff
@@ -115,50 +132,95 @@ def decompile(self, data, ttFont):
 		glyphOrder = ttFont.getGlyphOrder()
 		for k in range(nPairs):
 			left, right, value = next(it), next(it), next(it)
-			if value >= 32768: value -= 65536
+			if value >= 32768:
+				value -= 65536
 			try:
 				kernTable[(glyphOrder[left], glyphOrder[right])] = value
 			except IndexError:
-				# Slower, but will not throw an IndexError on an invalid glyph id.
-				kernTable[(ttFont.getGlyphName(left), ttFont.getGlyphName(right))] = value
-		if len(data) > 6 * nPairs + 4: # Ignore up to 4 bytes excess
-			log.warning("excess data in 'kern' subtable: %d bytes", len(data) - 6 * nPairs)
+				# Slower, but will not throw an IndexError on an invalid
+				# glyph id.
+				kernTable[(
+					ttFont.getGlyphName(left),
+					ttFont.getGlyphName(right))] = value
+		if len(data) > 6 * nPairs + 4:  # Ignore up to 4 bytes excess
```

Comment:
perhaps part of the responsibility for not noticing this bug before is that we are silently letting pass up to 4 extra bytes in the kern subtable. Why don't we always warn when the data length doesn't match the stated number of pairs or is not multiple of 6?

Reply:
so, Calibri has 4 bytes in excess.. https://github.com/fonttools/fonttools/commit/5cd0a55635062f382ff86ccea6e42bbf9b8b030a
ok then  :man_shrugging:

Reply:
Well, I still see Calibri's kern table shrink a _lot_ after roundtripping with ttx. I mentioned it in https://github.com/fonttools/fonttools/commit/5cd0a55635062f382ff86ccea6e42bbf9b8b030a but I don't think anyone followed up on that. Check `ttx -l` before and after roundtripping. But then again, I'm not entirely sure where that Calibri came from.

Reply:
I can send you the version I have if you're curious.

Reply:
Not particularly. I'd be more interested in having the AAT kern compilation fixed. Maybe you could also do a quick review since you wrote the original code? Thanks anyway :)

Reply:
sure, you can send me that file, thanks

Reply:
I see Win10 calibri.ttf has 160kb kern table.  I suppose MS started ignoring the binary search header completely and use the subtable length instead.  We should do the same.  I'll fix HarfBuzz as well.

Reply:
My bad.  Ignoring subtable length, respecting bsearch header!  This is now implemented in HarfBuzz:
https://github.com/behdad/harfbuzz/commit/82a38d1f7a65537a4ef540af08c489512d6297ac